conn.go

```diff
@@ -130,6 +132,58 @@ func (cc *Conn) netlinkConnUnderLock() (*netlink.Conn, netlinkCloser, error) {
 	return nlconn, func() error { return nlconn.Close() }, nil
 }
 
+func (cc *Conn) receiveAckAware(nlconn *netlink.Conn, sentMsgFlags netlink.HeaderFlags) ([]netlink.Message, error) {
+	if nlconn == nil {
+		return nil, errors.New("netlink conn is not initialized")
+	}
+
+	// first receive will be the message that we expect
+	reply, err := nlconn.Receive()
+	if err != nil {
+		return nil, err
+	}
+
+	if !cc.lasting {
+		// in non lasting connection there are no acks being sent,
+		// everything is a fresh connection
+		return reply, nil
+	}
+
+	if (sentMsgFlags & netlink.Acknowledge) == 0 {
+		// we did not request an ack
+		return reply, nil
+	}
+
+	if (sentMsgFlags & netlink.Dump) == netlink.Dump {
+		// sent message has Dump flag set, there wil be no acks
+		// https://github.com/torvalds/linux/blob/7e062cda7d90543ac8c7700fc7c5527d0c0f22ad/net/netlink/af_netlink.c#L2387-L2390
+		return reply, nil
+	}
+
+	// Dump flag is not set, we expect an ack
+	ack, err := nlconn.Receive()
+	if err != nil {
+		return reply, err
+	}
+
+	if len(ack) == 0 {
```

Comment:
Isn’t len(ack) == 0 an error case? I.e. we requested an ack but didn’t get one?

Reply:
Thanks, I was not sure what is the best approach to handle this since we've already got the reply that we need at this point and acks are not something that we handle other than receive and "ignore". In any case, we can notify that ack was empty but still return the original reply and let the user decide what to do with it.

Reply:
If the user is supposed to decide, we need to make the returned error a sentinel so folks can use errors.Is/errors.As. 

But, I don’t think it’s a good choice here to let the user decide. An empty ack sounds like a bug in our library and/or in the kernel, not something that one group of users might want while another group might not want.

I would favor returning just an error, and if we get reports of problems with that approach we can revisit.

Reply:
Done.